### DISCUSSION TITLE
Differences between JavaScript 32-Hour & Python 16-Hour.

### DISCUSSION THREAD

**Discussion by promenox** [smartcontractkit/full-blockchain-solidity-course-py#1806] [General]
Hello All:

Are there significant differences between the **JavaScript 32-Hour** & **Python 16-Hour** courses? 

I prefer Python - however, the 32-Hour course seems more inclusive.

Kind regards.

**Comment by promenox** [comment]
This post from **Patric Collins** answered my question:

[https://medium.com/@patrick.collins_58673/learn-blockchain-development-in-one-of-these-two-videos-c7db210b98fd](https://medium.com/@patrick.collins_58673/learn-blockchain-development-in-one-of-these-two-videos-c7db210b98fd)

**Reply by cromewar** [reply]
Nice, also remember that you will learn a lot on both of them so at the end is your personal preference.

**Comment by Niferu** [comment]
Hi @voxelfox 

I have already finished python course like month ago and now I'm in middle of JS course and I must say JS course is just amazing! It is much more clearer (maybe I just understand more now) and there is a lot more helpful information and tips. Hardhat is also greater tool than brownie in my opinion as it has more amazing features like for example gas-reporter. If you like python as I do you should definitely take a look at JS course anyway in my opinion. I promise you will not regret it :)

PS: @cromewar didn't lie to me as he was encouraging me to do JS course anyway and it is working great so far for me!

**Reply by cromewar** [reply]
Yes the JS version is totally worth it.
